### PR TITLE
Upgrade to latest stable version of yarn: 0.23.2

### PR DESCRIPTION
**Pull request by balderdash**

### CODE REVIEW

**Comment by pcardune** [comment]
Adding some install notes for posterity:

Everybody will need to update their version of yarn, as the format of `yarn.lock` changed ever so slightly. Depending on how they originally installed yarn, they will have to run one of the following commands:

**Ubuntu**
See the [yarn install docs](https://yarnpkg.com/lang/en/docs/install/) and run the following:

```bash
# Add the yarn repository per instructions in their docs and then run
sudo apt-get update
# Install the new specific version of yarn
sudo apt-get install yarn=0.23.2-1

# Tell the system not to auto-update yarn
sudo apt-mark hold yarn
```

**OS X**
```bash
brew unlink yarn
brew install yarn
```

**npm**
```bash
npm install -g yarn@0.23.2
```

**shell script**
```bash
curl -o- -L https://yarnpkg.com/install.sh | bash -s -- --version 0.23.2
```

Also, @wjordan, want to make sure you've taken a look at this change.

File: lib/rake/build.rake [hunk 1]
```diff
@@ -10,6 +10,9 @@ namespace :build do
       ChatClient.log 'Installing <b>apps</b> dependencies...'
       RakeUtils.npm_install
 
+      # Workaround for https://github.com/sass/node-sass/issues/1804
+      RakeUtils.npm_rebuild 'node-sass'
```
Comment (wjordan) [inline lib/rake/build.rake:14]: About how long does this workaround-step take to run? This runs on every single build, which could add a significant amount of additional time to every single step of our pipeline (most significantly launching new frontend instances, which we want to keep as short as possible for auto-scaling purposes). If it's < 1 minute it should be no problem, if it's > 1 minute let's try and find a smarter way of running this more selectively, as long as this workaround will be necessary moving forward.

Comment (balderdash) [reply lib/rake/build.rake:14]: It took 8 seconds in my circle run, and about the same locally.

**Comment by wjordan** [comment]
LGTM

**Comment by pcardune** [comment]
Once this gets merged, we should send an email out to dev for people to upgrade locally.

**Comment by islemaster** [comment]
Thanks for finding a solution to this!  I'm looking forward to being on recent `yarn`.

**Comment by balderdash** [comment]
I'll send out an email with basically a copy of Paul's comment.

**Pull request merged**